File: Java/Stack/DynamicStack.java
class DynamicStack {
    static int[] arr;
    static int top = 0;
    static int MAX;

    static boolean isEmpty() {
        return (top == 0);
    }

    static boolean isFull() {
        return (top == MAX);
    }

    static void push(int x) {
        if (isFull()) {
            System.out.println("Stack full");
        } else {
            arr[top] = x;
            top++;
            System.out.println(x + " pushed into stack.");
        }
    }

    static void pop() {
        if (isEmpty()) {
            System.out.println("Stack empty");
        } else {
            top--;
            System.out.println(arr[top] + " popped from stack.");
        }
    }

    public static void main(String[] args) {
        MAX = 5;
        arr = new int[MAX];

        push(10);
        push(20);
        pop();
        push(30);
    }
}

File: CPP/Stack/DynamicStack.cpp
#include <iostream>
using namespace std;

int *arr;
int top = 0;
int MAX;

bool isEmpty() {
    return (top == 0);
}

bool isFull() {
    return (top == MAX);
}

void push(int x) {
    if (isFull()) {
        cout << "Stack full" << endl;
    } else {
        arr[top] = x;
        top++;
        cout << x << " pushed into stack." << endl;
    }
}

void pop() {
    if (isEmpty()) {
        cout << "Stack empty" << endl;
    } else {
        top--;
        cout << arr[top] << " popped from stack." << endl;
    }
}

int main() {
    cout << "Enter stack size: ";
    cin >> MAX;

    arr = new int[MAX];

    push(10);
    push(20);
    pop();
    push(30);

    return 0;
}